app/lib/search/index_simple.dart

```diff
@@ -168,6 +161,80 @@ class SimplePackageIndex implements PackageIndex {
       value: (String url) => _documents[url].popularity * 100,
     );
   }
+
+  Score _searchText(String text, String packagePrefix) {
+    Score textScore;
+    if (text != null && text.isNotEmpty) {
+      textScore = new Score()
+        ..addValues(_nameIndex.search(text), 0.82)
+        ..addValues(_descrIndex.search(text), 0.12)
+        ..addValues(_readmeIndex.search(text), 0.06);
+      textScore.removeLowScores(0.05);
+    } else if (packagePrefix != null) {
+      textScore = new Score()..addValues(_nameIndex.search(packagePrefix), 0.8);
```

Comment:
Is this `_nameIndex.search(packagePrefix)` really searching for a **prefix** or does it match anywhere in the name?

If it doesn't search for a prefix, then we shouldn't use the name "prefix".

Reply:
Yeah, this was tricky to write (it is there in the older version too), and now it may not be needed at all. The idea was that if you don't specify any text query, yet specify a package name prefix, we still need to order the results by something, and at that point we only had this kind of "something".

I'll remove this, and disallow queries that have no text but filter on package prefix and order on text match score.